iceoryx_hoofs/README.md

```diff
@@ -39,112 +22,136 @@ and we do not use dynamic memory. In these cases we adjusted the API to our use
-|`variant`              |          | C++11 implementation of the C++17 feature `std::variant`                                                                                                                                                                              |
-|`variant_queue`        |          | A queue which wraps multiple variants of Queues (FiFo, SoFi, ResizeableLockFreeQueue)                                                                                                                                                 |
-|`vector`               |          | Heap and exception free implementation of `std::vector`                                                                                                                                                                               |
 
-### Concurrent
+### Primitives (primitives)
 
-If you have to write concurrent code, never use concurrency constructs like `mutex`, `atomic`, `thread`, `semaphore`, etc. directly. Most of the use cases can be solved by using an `ActiveObject` which uses as building block our `FiFo` or a
-queue which is thread-safe when combined with `smart_lock`. To learn more about active objects see [Prefer Using Active Objects Instead Of Naked Threads](https://www.drdobbs.com/parallel/prefer-using-active-objects-instead-of-n/225700095).
+| class                 | internal | description                                                                                                                                                                                                                           |
+|:---------------------:|:--------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|`type_traits`          |          | Extended support for evaluating types on compile-time.                                                                                                                                                                                |
+|`types`                |          | Declares essential building block types like `byte_t`.                                                                                                                                                                                |
+|`attributes`           |          | C++17 and C++20 attributes are sometimes available through compiler extensions. The attribute macros defined in here (like `IOX_FALLTHROUGH`, `IOX_MAYBE_UNUSED` ... ) make sure that we are able to use them if the compiler supports it. |
+|`algorithm`            |          | Implements `min` and `max` for an arbitrary number of values of the same type. For instance `min(1,2,3,4,5);`                                                                                                                         |
+
+### Queues (queues)
```

Comment:
The stack is in here so I really would rename this to Buffer. In my opinion everything with a push pop interface is a buffer and the lifo / fifo with queue and stack are just defining the buffer strategy/policy.

If you leave it like this people think we are stupid and do not know that a stack isn't a queue. :smile:

Reply:
Yeah, you are right. Buffers can have both FIFO and LIFO behaviour. FIFOs are also called queues and LIFOs also known as stacks. It still like the module name `PuPo` better :grin:

Reply:
Then it is decided it is called `PuPo` since it has a PushPop interface :smile: